src/Juvix/Compiler/Core/Data/TransformationId.hs

```diff
@@ -69,7 +69,10 @@ toVampIRTransformations = [CombineInfoTables, FilterUnreachable, CheckVampIR, Le
 
 toStrippedTransformations :: [TransformationId]
 toStrippedTransformations =
-  [CombineInfoTables, FilterUnreachable, CheckExec, LambdaLetRecLifting, TopEtaExpand, OptPhaseExec, MoveApps, RemoveTypeArgs]
+  [CombineInfoTables, FilterUnreachable, LambdaLetRecLifting, TopEtaExpand, OptPhaseExec, MoveApps, RemoveTypeArgs]
```

Comment:
The order was significant here -- "illegal" things (for `CheckExec` - axioms) may get filtered out or brought in from imported modules. Plus, there should be similar checks for Anoma/Nockma and later Cairo. I'll make a commit to this PR to refactor the transformation pipeline to allow for different kinds of checks.

Reply:
I see - we moved the CheckExec to a toExec transformation (https://github.com/anoma/juvix/blob/8f7a6426955d2fc893c12dc2ac58c0ef5ca322ba/src/Juvix/Compiler/Pipeline.hs#L178) to allow Anoma main functions to have arbitrary signatures. But in the case of Anoma we want to check that axioms are not present so I think your refactor is a good idea, thanks.

Reply:
I commited a refactor.